snapshot/btrfs/btrfs.go

```diff
@@ -23,7 +25,9 @@ func init() {
 	plugin.Register(&plugin.Registration{
 		ID:   "btrfs",
 		Type: plugin.SnapshotPlugin,
-		Init: func(ic *plugin.InitContext) (interface{}, error) {
+		InitFn: func(ic *plugin.InitContext) (interface{}, error) {
+			ic.Meta.Platforms = []ocispec.Platform{platforms.DefaultSpec()}
```

Comment:
[I'm not sure ite makes sense to add platform info to snapshotter services.
Isn't it agnostic to any platform?

Reply:
How is a btrfs not agnostic to the platform? It requires linux.

Reply:
At least, it doesn't require the architecture and variant specified by `platforms.DefaultSpec()`.

Reply:
Absolutely, it does. This is for the snapshotter, not the GRPC service. If you are running two snapshotters for different platforms, there is no reason to expect them to work together.

Reply:
Mmm. Could you elaborate what `platforms` actually stands for?
Probably I'm mistaking the meaning of this.

I thought AMD64 snapshotter can manipulate rootfs snapshots for PPC64LE images.
Although AMD64 runtime is unlikely to be able to execute PPC64LE containers, but it is correct usecase to manipulate snapshotters without runtime.

Reply:
@AkihiroSuda If the snapshotters support all those platforms on a single machine, then they can explicitly state that, but support is not necessarily a given. Most implementations will be bound to the architecture that the host can support.

Reply:
If this field is about runtime aspect, I'd suggest s/Platform/RuntimePlatform/.

WDYT?

Reply:
@AkihiroSuda No, we are not specializing for each plugin. The snapshotters, and all plugins that declare a platform, are platform-specific. I don't see why you want to complicate this model. Could you at least read the data structure documentation?

Reply:
Multi-platform support would need to be implemented as separate instances, having a single platform for each instance is correct though. This is code is accurate since it represents the current support for the plugin.

Reply:
I read through the code and docs, but still confused about how client apps are expected to utilize this platform info.

ex. For following use cases, what should client apps do?

- Image builder app  running on `linux/amd64` host: Copy files under a `linux/ppc64le` image snapshot to a `linux/amd64` image snapshot
- Image vulnerability scanner app running on `linux/amd64` host:  Statically scan files under a `linux/ppc64le` image snapshot


I agree we need to put OS info (windows/linux) here for LCOW support, but can we omit `architecture` and `variant` from snapshotter plugins? (We can add it later on demand)